llvm/lib/Analysis/LoopAccessAnalysis.cpp

```diff
@@ -1885,17 +1885,9 @@ MemoryDepChecker::isDependent(const MemAccessInfo &A, unsigned AIdx,
 
   // If the induction step is negative we have to invert source and sink of the
```

Comment:
This comment is no accurate any longer? Now just the distance is inverted, right? 

Src/sink will get printed in the original order which may be confusing?

Reply:
I intentionally left the comment since we are still inverting the src/sink parameters to getMinusSCEV(). Do you want me to rephrase it perhaps? The print order is the program order which the assertion a few lines above checks:
 
`assert (AIdx < BIdx && "Must pass arguments in program order");`

Reply:
I've updated the comment and left the printing order unchanged (same as program order).

Reply:
IIUC we should still swap source and sink for printing, as otherwise it may be confusing that the difference is not what we expect (Sink - Src)?

Would be good to also add a test for the debug output.

Reply:
Any thoughts on the above?

Reply:
I've added tests and inverted the print order.